File: pfe-java/app/controllers/Application.java
package controllers;

import play.Routes;
import play.cache.Cached;
import play.mvc.Result;
import play.twirl.api.JavaScript;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public class Application extends Controller {

    @Inject
    public Application(Service service) {
        super(service);
    }

    @Cached(key = "main-html")
    public Result index() {
        return ok(views.html.main.render());
    }

    public Result javascriptRouter() {
        JavaScript router = Routes.javascriptRouter("routes",
                routes.javascript.Items.delete(),
                routes.javascript.Auctions.bid(),
                routes.javascript.Auctions.notifications(),
                routes.javascript.Auctions.channel()
        );
        return ok(JavaScript.apply("define(function () { " + router.body() + "; return routes })"));
    }

}

File: pfe-scala/app/controllers/Application.scala
package controllers

import play.api.mvc.{Controller, Action}
import play.api.routing.JavaScriptReverseRouter
import play.twirl.api.JavaScript
import play.api.cache.Cached
import play.api.i18n.{Messages, I18nSupport, MessagesApi, Lang}

class Application(cached: Cached, val messagesApi: MessagesApi) extends Controller with I18nSupport {

  val index = cached(implicit request => s"main-html-${implicitly[Messages].lang.code}") {
    Action { implicit request =>
      Ok(views.html.main())
    }
  }

  def javascriptRouter = Action { implicit request =>
    val router = JavaScriptReverseRouter("routes")(
      routes.javascript.Items.delete,
      routes.javascript.Auctions.bid,
      routes.javascript.Auctions.notifications,
      routes.javascript.Auctions.channel
    )
    Ok(JavaScript(s"""define(function () { $router; return routes })"""))
  }

}
